app/src/main/java/in/testpress/testpress/authenticator/CodeVerificationActivity.java

```diff
@@ -0,0 +1,194 @@
+package in.testpress.testpress.authenticator;
+
+import android.app.Activity;
+import android.app.AlertDialog;
+import android.app.Dialog;
+import android.app.ProgressDialog;;
+import android.content.DialogInterface;
+import android.content.Intent;
+import android.os.Bundle;
+import android.text.Editable;
+import android.text.TextWatcher;
+import android.view.Gravity;
+import android.view.KeyEvent;
+import android.view.View;
+import android.widget.Button;
+import android.widget.EditText;
+import android.widget.TextView;
+
+import javax.inject.Inject;
+
+import butterknife.ButterKnife;
+import butterknife.InjectView;
+import butterknife.OnClick;
+import in.testpress.testpress.Injector;
+import in.testpress.testpress.R;
+import in.testpress.testpress.core.TestpressService;
+import in.testpress.testpress.models.RegistrationSuccessResponse;
+import in.testpress.testpress.models.RegistrationErrorDetails;
+import in.testpress.testpress.ui.MainActivity;
+import in.testpress.testpress.ui.TextWatcherAdapter;
+import in.testpress.testpress.util.SafeAsyncTask;
+import retrofit.RetrofitError;
+
+import static android.view.inputmethod.EditorInfo.IME_ACTION_DONE;
+
+public class CodeVerificationActivity extends Activity {
+    @Inject TestpressService testpressService;
+    @InjectView(R.id.welcome) TextView welcomeText;
+    @InjectView(R.id.et_username) EditText usernameText;
+    @InjectView(R.id.et_verificationCode) EditText verificationCodeText;
+    @InjectView(R.id.b_verify) Button verifyButton;
+
+    private String username;
+    private String password;
+    private boolean unknownUser;
+    private RegistrationSuccessResponse codeResponse;
+    private final TextWatcher watcher = validationTextWatcher();
+
+    @Override
+    public void onCreate(Bundle bundle) {
+        super.onCreate(bundle);
+        Injector.inject(this);
+        setContentView(R.layout.code_verify_activity);
+        ButterKnife.inject(this);
+        final Intent intent = getIntent();
+        username=intent.getStringExtra("username");
+        password=intent.getStringExtra("password");
+        unknownUser=username==null;
+        if(unknownUser) {
+            usernameText.setVisibility(View.VISIBLE);
+        }
+        else
+            welcomeText.setText("Welcome "+username);
```

Comment:
Coding guidelines incorrect. else should be in same line as the previous closing bracket and should have braces even if single line code is present.

    } else {
      welcomeText.setText("Welcome "+username);
    }